src/coreclr/inc/corinfo.h

```diff
@@ -1728,10 +1728,16 @@ struct CORINFO_FIELD_INFO
 
 struct CORINFO_THREAD_STATIC_BLOCKS_INFO
 {
+#ifdef _MSC_VER
```

Comment:
We cannot have any platform-specific ifdefs in JIT/EE interface. It breaks cross-compilation.

Reply:
Sure, just that I understand, the fields should be same and shouldn't be different depending on the platform, right?

Reply:
Are you suggesting something like having unused fields without having `#ifdef` ?

```c++
struct CORINFO_THREAD_STATIC_BLOCKS_INFO
{
    CORINFO_CONST_LOOKUP tlsIndex;              // on linux, it will be unused
    uint32_t offsetOfThreadLocalStoragePointer; // on linux, it will be unused
    uint32_t offsetOfMaxThreadStaticBlocks;     // on linux this will have descriptor address
    uint32_t offsetOfThreadStaticBlocks; 
    size_t tlsGetAddrFtnPtr;                   // on windows, it will be unused
    uint32_t offsetOfGCDataPointer;
};

```

Reply:
> This cannot have descriptor address

ah yes, I deleted the pointer-sized field and forgot to update it here. Makes sense.

Reply:
Fixed.